test/functional/mempool_limit.py

```diff
@@ -34,51 +34,224 @@ def set_test_params(self):
         # And 2 more for the package cpfp test
-        self.generate(miniwallet, 1 + (num_of_batches * tx_batch_size) + 1 + 2)
+        self.generate(miniwallet, 1 + (num_of_batches * tx_batch_size))
 
         # Mine 99 blocks so that the UTXOs are allowed to be spent
         self.generate(node, COINBASE_MATURITY - 1)
 
-        self.log.info('Create a mempool tx that will be evicted')
+        self.log.debug("Create a mempool tx that will be evicted")
         tx_to_be_evicted_id = miniwallet.send_self_transfer(from_node=node, fee_rate=relayfee)["txid"]
 
         # Increase the tx fee rate to give the subsequent transactions a higher priority in the mempool
         # The tx has an approx. vsize of 65k, i.e. multiplying the previous fee rate (in sats/kvB)
         # by 130 should result in a fee that corresponds to 2x of that fee rate
         base_fee = relayfee * 130
 
-        self.log.info("Fill up the mempool with txs with higher fee rate")
-        for batch_of_txid in range(num_of_batches):
-            fee = (batch_of_txid + 1) * base_fee
-            create_lots_of_big_transactions(miniwallet, node, fee, tx_batch_size, txouts)
+        self.log.debug("Fill up the mempool with txs with higher fee rate")
+        with node.assert_debug_log(["rolling minimum fee bumped"]):
+            for batch_of_txid in range(num_of_batches):
+                fee = (batch_of_txid + 1) * base_fee
+                create_lots_of_big_transactions(miniwallet, node, fee, tx_batch_size, txouts)
 
-        self.log.info('The tx should be evicted by now')
+        self.log.debug("The tx should be evicted by now")
         # The number of transactions created should be greater than the ones present in the mempool
         assert_greater_than(tx_batch_size * num_of_batches, len(node.getrawmempool()))
         # Initial tx created should not be present in the mempool anymore as it had a lower fee rate
         assert tx_to_be_evicted_id not in node.getrawmempool()
 
-        self.log.info('Check that mempoolminfee is larger than minrelaytxfee')
+        self.log.debug("Check that mempoolminfee is larger than minrelaytxfee")
         assert_equal(node.getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
         assert_greater_than(node.getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
 
+    def test_mid_package_eviction(self):
+        node = self.nodes[0]
+        self.log.info("Check a package where each parent passes the current mempoolminfee but would cause eviction before package submission terminates")
+
+        self.restart_node(0, extra_args=self.extra_args[0])
+
+        # Restarting the node resets mempool minimum feerate
+        assert_equal(node.getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
+        assert_equal(node.getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
+
+        self.fill_mempool()
```

Comment:
As I understand one of the condition to hit the case is a full mempool and when we call `LimitMempool()`, one interesting variation of the test case could be to limit the mempool size to the _subpackage_ only. It might be a tricky one, though boundary conditions are always nice to exercise.

Reply:
Yes, full mempool is one of the conditions (i.e. the one tested in `test_mid_package_eviction`). It's not strictly necessary though, the coin can also disappear due to replacement.

> one interesting variation of the test case could be to limit the mempool size to the subpackage only

Do you mean setting the mempool's maximum memory to something very tiny?